quixstreams/models/topics.py

```diff
@@ -28,7 +33,79 @@
 
 logger = logging.getLogger(__name__)
 
-__all__ = ("Topic",)
+__all__ = ("Topic", "TopicKafkaConfigs")
+
+
+class TopicKafkaConfigs:
+    _quix_optionals_defaults = {
+        "retention.ms": f"{10080 * 60000}",  # minutes converted to ms
+        "retention.bytes": "52428800",
+    }
+
+    def __init__(
+        self,
+        name: Optional[str] = None,  # Required when not created by a Quix App.
+        num_partitions: Optional[int] = None,
+        replication_factor: Optional[int] = None,
+        optionals: Optional[dict] = None,
+        is_quix_topic: bool = False,
+    ):
+        # TODO: make as properties
+        self._is_quix_topic = is_quix_topic
+        self.name = name
+        self.num_partitions = num_partitions or 2
+        self.replication_factor = replication_factor or (2 if is_quix_topic else 1)
+        self._optionals = optionals or {}
+
+    @property
+    def is_quix_topic(self) -> bool:
+        return self._is_quix_topic
+
+    @property
+    def optionals(self) -> Dict[str, str]:
+        return self._optionals
+
+    def _set_quix_optional_defaults(self):
+        for k in self._quix_optionals_defaults:
+            self._optionals.setdefault(k, self._quix_optionals_defaults[k])
+
+    def _filter_quix_optionals(self):
+        invalid_optionals = [
+            k for k in self.optionals if k not in self._quix_optionals_defaults
+        ]
+        self._optionals = {
+            k: v
+            for k, v in self.optionals.items()
+            if k in self._quix_optionals_defaults
+        }
+        if invalid_optionals:
+            warnings.warn(
+                f"There are optionals provided that are not supported by "
+                f"the Quix API and thus will be removed from the optionals:"
+                f"{invalid_optionals}"
+            )
+
+    def _as_quix_create(self) -> Self:
+        new = deepcopy(self)
+        new._filter_quix_optionals()
+        new._set_quix_optional_defaults()
+        return new
+
+    def _as_create(self) -> NewTopic:
+        if self.is_quix_topic:
+            raise Exception(
```

Comment:
The exception class must be specific. Raising plain `Exception` is not a good practice